corehq/apps/reports/templates/reports/reportdata/case_delete.html

```diff
@@ -0,0 +1,160 @@
+{% extends "hqwebapp/bootstrap3/base_section.html" %}
+{% load case_tags %}
+{% load hq_shared_tags %}
+{% load i18n %}
+{% load proptable_tags %}
+{% load timezone_tags %}
+
+{% block head %} {{ block.super }}
+  <link rel="stylesheet" type="text/css" href="{% static "hqwebapp/css/proptable.css" %}">
+{% endblock %}
+
+{% requirejs_main 'reports/js/case_details' %}
+
+{% block page_content %}
+  <div>
+    <h2>
+      Case Deletion
+    </h2>
+    <h3 class="h4">
+      Please review the listed items before proceeding with the case deletion.
+    </h3>
+    </br>
+    {% blocktrans %}
+      The following Cases and their Form Submissions will be deleted:
+    {% endblocktrans %}
+    <ul>
+      {% for case_link, forms in delete_dict.items %}
+        <li>
+          <h4><i class="fa fa-list-alt"></i><strong>{{ case_link }}</strong></h4>
+            <ul class="list-group">
+              {% for form_link, form_actions in forms.items %}
+              <li class="list-group-item">
+                <h5><i class="fa fa-file"></i>{{ form_link }}</h5>
+                Actions taken by this form on:
+                <ul>
+                  {% for case_name, actions in form_actions.items %}
+                    <li>
+                      {% if case_name == 'current' %}
+                        This case: {{ actions }}
+                      {% else %}
+                        Related case: <strong>{{ case_name }}</strong>: {{ actions }}
+                      {% endif %}
+                    </li>
+                  {% endfor %}
+                </ul>
+              </li>
+              {% endfor %}
+            </ul>
+        </li>
+      {% endfor %}
+    </ul>
+    </br>
+    {% if affected_cases %}
+      {% blocktrans %}
+        The following Cases have related Form Submissions that will be deleted, which will reverse the action
+        taken by that form, on the case. These cases will <strong>not</strong> be deleted.
+      {% endblocktrans %}
+      <ol>
+        {% for case_link, forms in affected_cases.items %}
+          <li>
+            <h5><strong>{{ case_link }}</strong></h5>
+            <ul>
+              {% for form_link, form_actions in forms.items %}
+                <li>
+                  <i class="fa fa-file"> </i>
+                  {{ form_link }} ({{ form_actions }})
+                </li>
+              {% endfor %}
+            </ul>
+          </li>
+        {% endfor %}
+      </ol>
+    {% endif %}
+    </br>
+    {% if reopened_cases %}
+      {% blocktrans %}
+        The following Cases will be reopened but <strong>not</strong> deleted.
+      {% endblocktrans %}
+      <ol>
+      {% for case_link, form_link in reopened_cases.items %}
```

Comment:
I see the `case_link` html is being generated in python above. can these instead just be URLs and the link HTML is specified in the template? It's always better to separate design from function.